src/main/java/org/radarcns/redcap/util/RedCapClient.java

```diff
@@ -0,0 +1,160 @@
+package org.radarcns.redcap.util;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.*;
+
+import okhttp3.*;
+import org.json.JSONArray;
+import org.radarcns.redcap.config.RedCapInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/*
+ * Copyright 2017 King's College London
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Skeleton for creating a REDCap update handler.
+ */
+@SuppressWarnings("unchecked")
+public class RedCapClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RedCapClient.class);
+    private static final String API_ROOT = "/redcap/api/";
+    private static final String TOKEN_LABEL = "token";
+    private static final String DATA_LABEL = "data";
+    private static final String FIELDS_LABEL = "fields";
+    private static final String RECORDS_LABEL = "records";
+    private static final int REDCAP_RESULT_INDEX = 0;
+
+
+    private final RedCapInfo redCapInfo;
+
+    public final OkHttpClient httpClient = new OkHttpClient();
```

Comment:
any reason this is public?

Reply:
No specific reason, made it private now.